### PR TITLE
*: add tidb_opt_use_invisible_indexes

### PR DESCRIPTION
**Pull request by hawkingrei**
### First-time contributors' checklist 

- [ ] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs-cn) that's required for repo owners to accept my contribution.

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions (in Chinese)](https://github.com/pingcap/docs-cn/blob/master/CONTRIBUTING.md#版本选择指南).

- [x] master (the latest development version)
- [x] v8.0 (TiDB 8.0 versions)
- [ ] v7.6 (TiDB 7.6 versions)
- [ ] v7.5 (TiDB 7.5 versions)
- [ ] v7.4 (TiDB 7.4 versions)
- [ ] v7.1 (TiDB 7.1 versions)
- [ ] v6.5 (TiDB 6.5 versions)
- [ ] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from:

https://github.com/pingcap/docs/pull/16762

- Other reference link(s):

https://github.com/pingcap/tidb/pull/50654

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

**Review by AilinKid — APPROVED** [review]

File: system-variables.md [hunk 1]
```diff
@@ -3850,6 +3850,14 @@ SHOW WARNINGS;
 - 默认值：`24.0`
 - 表示 TiFlash 计算的并发数。该变量是[代价模型](/cost-model.md)内部使用的变量，**不建议**修改该变量的值。
 
+## `tidb_opt_use_invisible_indexes`
```
Comment (Oreoxmt) [inline system-variables.md:3853]:
```suggestion
## `tidb_opt_use_invisible_indexes` <span class="version-mark">从 v8.0.0 版本开始引入</span>
```

File: system-variables.md [hunk 2]
```diff
@@ -3850,6 +3850,14 @@ SHOW WARNINGS;
...
+
+- 作用域：SESSION
+- 是否受 Hint [SET_VAR](/optimizer-hints.md#set_varvar_namevar_value) 控制：是
+- 类型：布尔型
+- 默认值：`OFF`
+- 这个变量用来设置是否允许优化器使用隐藏索引。隐藏索引是指用户创建的索引，但是该索引并不会被优化器使用。开启该选项后，优化器会尝试使用隐藏索引。
```
Comment (Oreoxmt) [inline system-variables.md:3859]:
```suggestion
- 这个变量用来设置是否允许优化器选择[不可见索引 (Invisible Index)](/sql-statements/sql-statement-create-index.md#不可见索引)。默认情况下，不可见索引由 DML 语句维护，不会被查询优化器使用。当修改变量为 `ON` 时，对该会话中的查询，优化器可以选择不可见索引进行查询优化。
```

Comment (Oreoxmt) [reply system-variables.md:3859]: 请修改以下不可见索引相关的文档描述，例如添加 tidb_opt_use_invisible_indexes 系统变量的描述

https://github.com/pingcap/docs-cn/blob/874e88785efd909bd9a9ea14ae80761231ad9af4/sql-statements/sql-statement-alter-index.md?plain=1#L9

https://github.com/pingcap/docs-cn/blob/874e88785efd909bd9a9ea14ae80761231ad9af4/sql-statements/sql-statement-alter-index.md?plain=1#L122

https://github.com/pingcap/docs-cn/blob/874e88785efd909bd9a9ea14ae80761231ad9af4/sql-statements/sql-statement-create-index.md?plain=1#L387

File: sql-statements/sql-statement-alter-index.md [hunk 3]
```diff
@@ -6,7 +6,7 @@ aliases: ['/docs-cn/dev/sql-statements/sql-statement-alter-index/','/docs-cn/dev
 
 # ALTER INDEX
 
-`ALTER INDEX` 语句用于修改索引的可见性，可以将索引设置为 `Visible` 或者 `Invisible`。设置为 `Invisible` 的索引即不可见索引 (Invisible Index) 由 DML 语句维护，不会被查询优化器使用。
+`ALTER INDEX` 语句用于修改索引的可见性，可以将索引设置为 `Visible` 或者 `Invisible`。设置为 `Invisible` 的索引即不可见索引 (Invisible Index) 由 DML 语句维护，不会被查询优化器使用。但是可以通过改变系统参数 [`tidb_opt_use_invisible_indexes`](/system-variables.md#tidb_opt_use_invisible_indexes) 参数，强制优化器使用不可见索引。
```
Comment (Oreoxmt) [inline sql-statements/sql-statement-alter-index.md:9]:
```suggestion
`ALTER INDEX` 语句用于修改索引的可见性，可以将索引设置为 `Visible` 或者 `Invisible`。设置为 `Invisible` 的索引即不可见索引 (Invisible Index) 由 DML 语句维护，不会被查询优化器使用。从 TiDB v8.0.0 开始，你可以通过修改系统变量 [`tidb_opt_use_invisible_indexes`](/system-variables.md#tidb_opt_use_invisible_indexes)，允许优化器选择不可见索引。
```

File: sql-statements/sql-statement-create-index.md [hunk 4]
```diff
@@ -391,6 +391,8 @@ CREATE TABLE t1 (c1 INT, c2 INT, UNIQUE(c2));
 CREATE UNIQUE INDEX c1 ON t1 (c1) INVISIBLE;
 ```
```
Comment (Oreoxmt) [inline sql-statements/sql-statement-create-index.md:393]: L387: 

```markdown
默认情况下，不可见索引 (Invisible Indexes) 不会被查询优化器使用：
```

File: sql-statements/sql-statement-create-index.md [hunk 5]
```diff
@@ -391,6 +391,8 @@ CREATE TABLE t1 (c1 INT, c2 INT, UNIQUE(c2));
...
 
+另外，通过改变系统参数 [`tidb_opt_use_invisible_indexes`](/system-variables.md#tidb_opt_use_invisible_indexes) 参数，强制优化器使用不可见索引。
```
Comment (Oreoxmt) [inline sql-statements/sql-statement-create-index.md:394]:
```suggestion
从 TiDB v8.0.0 开始，你可以通过修改系统变量 [`tidb_opt_use_invisible_indexes`](/system-variables.md#tidb_opt_use_invisible_indexes)，允许优化器选择不可见索引。
```

**Review by Oreoxmt — APPROVED** [review]

File: system-variables.md [hunk 6]
```diff
@@ -3850,6 +3850,14 @@ SHOW WARNINGS;
 - 默认值：`24.0`
 - 表示 TiFlash 计算的并发数。该变量是[代价模型](/cost-model.md)内部使用的变量，**不建议**修改该变量的值。
 
+## `tidb_opt_use_invisible_indexes` <span class="version-mark">从 v8.0.0 版本开始引入</span>
+
+- 作用域：SESSION
```
Comment (hfxsd) [inline system-variables.md:3855]:
```suggestion
- 作用域：SESSION
- 是否持久化到集群：
```

Comment (Oreoxmt) [reply system-variables.md:3855]: GLOBAL 作用域的需要说明是否持久化到集群
image

**Review by hfxsd — APPROVED** [review]

**Pull request merged**